### PR TITLE
Feature/community

### PR DESCRIPTION
**Pull request by goldentg**
Created a new community class for commands relating to community/fun usage. Commands i included are:

| Command Name | Command Purpose                                                                                                                                        |
| :----------: | ------------------------------------------------------------------------------------------------------------------------------------------------------ |
|  hug | Allows a user to hug another guildmember  `!hug <@user>`  |
| 8ball | Allows a user to ask a magic 8-ball for advise `!8ball {question}` |
| coinflip | Flips a coin `!coinflip` | 
| poll | allows a member to create a reaction based poll `!poll {poll message}` | 

Notes:
I noticed that previous code style involved `.this` statements but my code editor automatically removed them on save every time and despite spending 2 hours trying to remove that setting i was unable to locate the setting to do so.

If you have any revisions i would be happy to modify my code to meet the standards

### CODE REVIEW

File: CWE/CWE.csproj [hunk 1]
```diff
@@ -15,6 +15,10 @@
     <PackageReference Include="Discord.Net.Labs.WebSocket" Version="2.3.8" />
     <PackageReference Include="Discord.Net.WebSocket" Version="2.2.0" />
     <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.5" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="5.0.5">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
```
Comment (hendrikdemir) [inline CWE/CWE.csproj:21]: Why was this added if there were no adjustments made to the database?

File: CWE/CWE.csproj [hunk 2]
```diff
@@ -28,7 +32,7 @@
 
   <ItemGroup>
     <None Update="appsettings.json">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (hendrikdemir) [inline CWE/CWE.csproj:35]: Why was this adjusted?

File: CWE/Common/Colors.cs [hunk 3]
```diff
@@ -1,41 +1,63 @@
...
 
         /// <summary>
-        /// The color used to indicate an active state.
+        ///     The color used to indicate a new community poll.
         /// </summary>
-        public static Color Active = new Color(254, 200, 16);
+        public static Color Poll = new(33, 176, 252);
     }
-}
+}
```
Comment (hendrikdemir) [inline CWE/Common/Colors.cs:63]: This file contains way too many modifications, the only thing that should be different is the addition of i.e. colours for certain commands. Also, as these colours are only used within those commands, there is no need to add them to this class.

File: CWE/Common/Embeds.cs [hunk 4]
```diff
@@ -132,77 +133,140 @@ public static Embed GetRequestEmbed(Request request)
...
+                .WithThumbnailUrl(
+                    Icons.Poll)
+                .WithCurrentTimestamp()
+                .WithFooter($"Poll started by {user.Username + "#" + user.Discriminator}")
+                .Build();
+            return embed;
+        }
     }
-}
+}
```
Comment (hendrikdemir) [inline CWE/Common/Embeds.cs:272]: File again contains way too many changes, as again these embeds are only used in their respective commands, there is no need to add them to this class.

File: CWE/Common/Icons.cs [hunk 5]
```diff
@@ -1,69 +1,95 @@
...
+        public static string Coinflip = "http://clipart-library.com/data_images/296207.png";
+
+        /// <summary>
+        ///     The icon used to represent a poll.
+        /// </summary>
+        public static string Poll =
+            "https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.pinclipart.com%2Fpicdir%2Fmiddle%2F107-1071607_polls-icon-png-clipart.png&f=1&nofb=1";
     }
-}
+}
```
Comment (hendrikdemir) [inline CWE/Common/Icons.cs:95]: Again, lots of changes, icons of these commands don't need to be here.

File: CWE/Modules/Community.cs [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            var emoteUp = new Emoji("⬆️");
+            var emoteDown = new Emoji("⬇️");
+
+            var poll = Embeds.pollEmbed(pollmsg, Context.User);
+            var pollemb = await Context.Channel.SendMessageAsync(embed: poll);
+            await pollemb.AddReactionAsync(emoteUp);
+            await pollemb.AddReactionAsync(emoteDown);
+        }
+    }
+}
```
Comment (hendrikdemir) [inline CWE/Modules/Community.cs:137]: Not in accordance with StyleCop.Analyzers (i.e. position of using directives). Weird spacing before the summary, why do commands run using `RunMode.Async` when they are quite simple?

**Review by hendrikdemir — CHANGES_REQUESTED** [review]
There are a lot of changes added to existing classes that are unnecessary. For the commands you've made, there is no need to add i.e. the colours, icons and embeds to the existing classes, you can simply just put them all in the commands themselves. We only do this to prevent copying ourselves over and over. 

Also, I am not sure if this is something that we are really looking for. But, if you can go over the requested changes and fix the issue with the missing `this.`, I will be happy to review it again.

File: CWE/CWE.csproj (hunk 1 above)
Comment (goldentg) [reply CWE/CWE.csproj:21]: would not let me update database on my end in order to test the bot without it

**Comment by goldentg** [comment]
i will modify the code to comply with these requests later

File: CWE/CWE.csproj (hunk 1 above)
Comment (n-Ultima) [reply CWE/CWE.csproj:21]: This is indeed true. For me to run 'Update-Databade', it was required that the main CWE project had this NuGet package installed.

Comment (hendrikdemir) [reply CWE/CWE.csproj:21]: Make sure you set the "Default project" to CWE.Data :)

Comment (n-Ultima) [reply CWE/CWE.csproj:21]: Ah, never thought of that.

**Comment by hendrikdemir** [comment]
Hello there! I've thought a while about this PR but must conclude that these commands aren't commands that the community would benefit from. Surely, they can increase activity, but our server and ad hoc subject isn't about entertainment. Thank you for your PR, I really appreciate your initiative.
